### PR TITLE
Ngfw 14841 password encryption pppoe

### PR DESCRIPTION
**Pull request by VarshaKache**
**Issue**
For PPPoE interfaces, the password is stored in plain text format in the network.js file.
(Config -- Network -- Interface)

**Requirement:** 
Passwords should not be stored in plain text in the network.js file for PPPoE interfaces.

**Implementation:**
Using the password encryption and decryption utility added in the https://github.com/untangle/ngfw_src/pull/872 , the password is encrypted before being saved in the network file and decrypted to be displayed on the UI.

image

**Previous structure of json file for PPPoE:**

image

**Current structure of json file for for PPPoE:**

image

On the UI, the original password will be made visible by encrypting and decrypting it while it is masked.
image

### CODE REVIEW

File: uvm/impl/com/untangle/uvm/NetworkManagerImpl.java [hunk 1]
```diff
@@ -1788,6 +1789,13 @@ private void sanitizeNetworkSettings( NetworkSettings networkSettings )
                 intf.setV4PPPoERootDev(intf.getSystemDev());
                 intf.setSystemDev("ppp" + pppCount);
                 intf.setSymbolicDev("ppp" + pppCount);
+                //Encrypt the password for v4PPPoEPassword and save it in v4PPPoEPasswordEncrypted
+                if(StringUtils.isNotBlank(intf.getV4PPPoEPassword())){
+                    intf.setV4PPPoEPasswordEncrypted(PasswordUtil.getEncryptPassword(intf.getV4PPPoEPassword()));
+                }
+                if(StringUtils.isNotBlank(intf.getV4PPPoEPasswordEncrypted())){
```
Comment (mayur-dhande) [inline uvm/impl/com/untangle/uvm/NetworkManagerImpl.java:1796]: Do you need to check again notblank on `v4PPPoEPasswordEncrypted` ?

**Review by mayur-dhande — COMMENTED** [review]
Small feedback

File: uvm/impl/com/untangle/uvm/NetworkManagerImpl.java (hunk 1 above)
Comment (VarshaKache) [reply uvm/impl/com/untangle/uvm/NetworkManagerImpl.java:1796]: If password is already encrypted, then also we need to keep the v4PPPoEPassword as empty.
Other wise the v4PPPoEPassword field get removed from network.js file

**Review by mayur-dhande — APPROVED** [review]
Looks good, @DhirajM09 please review from your side too.

**Review by DhirajM09 — APPROVED** [review]

**Review by cblaise-untangle — APPROVED** [review]

**Pull request merged**